services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -102,31 +140,58 @@
 
     :raises web.HTTPNotFound: cannot find project id in repository
     """
-    project_uuid, uid = request.match_info.get("project_id"), request.get(RQT_USERID_KEY, ANONYMOUS_UID)
 
-    new_values = await request.json()
+    user_id = request[RQT_USERID_KEY]
+    project_uuid = request.match_info.get("project_id")
+    new_project = await request.json()
+
+    db = request.config_dict[APP_PROJECT_DBAPI]
+
+    await check_permission(request, "project.update | project.workbench.node.inputs.update",
+    context={
+        'dbapi': db,
+        'project_id': project_uuid,
+        'user_id': user_id,
+        'new_data': new_project
+    })
+
     try:
-        _validate(request.app, new_values)
+        validate_project(request.app, new_project)
+
+        await db.update_user_project(new_project, user_id, project_uuid)
+
     except ValidationError:
         raise web.HTTPBadRequest
 
-    try:
-        await ProjectDB.update_user_project(new_values, uid, project_uuid, db_engine=request.app[APP_DB_ENGINE_KEY])
     except ProjectNotFoundError:
         raise web.HTTPNotFound
 
+    return {'data': new_project}
+
+
+# TODO: temporary hidden until get_handlers_from_namespace refactor to seek marked functions instead!
+#@login_required
+#async def patch_project(_request: web.Request):
+#    """
+#        Client sends a patch and return updated project
+#        PATCH
+#    """
+#    # TODO: implement patch with diff as body!
+#    raise NotImplementedError()
+
+
 @login_required
 async def delete_project(request: web.Request):
-    project_uuid, uid = request.match_info.get("project_id"), request.get(RQT_USERID_KEY, ANONYMOUS_UID)
+    # TODO: replace by decorator since it checks again authentication
+    await check_permission(request, "project.delete")
+
+    user_id = request[RQT_USERID_KEY]
+    project_uuid = request.match_info.get("project_id")
+    db = request.config_dict[APP_PROJECT_DBAPI]
 
     try:
-        await ProjectDB.delete_user_project(uid, project_uuid, db_engine=request.app[APP_DB_ENGINE_KEY])
+        await db.delete_user_project(user_id, project_uuid)
```

Comment:
this should also delete all the dynamic services used by this project when this happens...